src/apps/vpn/ui/screens/getHelp/developerMenu/ViewDeveloperMenu.qml

```diff
@@ -64,18 +64,20 @@ MZViewBase {
                 if (MZSettings.stagingServerAddress !== serverAddressInput.text) {
                     MZSettings.stagingServerAddress = serverAddressInput.text;
                 }
-            }
-
-            Component.onCompleted: {
-                serverAddressInput.text = MZSettings.stagingServerAddress;
-            }
 
-            Component.onDestruction: {
-                // If a server address is not specified, don't use a staging server
+                // If the server address has been deleted, uncheck checkBoxRowStagingServer to not use a
+                // staging server. Also reset the server address to the default, so the user can easily switch
+                // back to it by rechecking the checkbox.
                 if (serverAddressInput.text.length == 0) {
                     MZSettings.stagingServer = false;
+                    serverAddressInput.text = "https://stage.guardian.nonprod.cloudops.mozgcp.net";
```

Comment:
Resetting to the default URL this way is a bit of a hack, because the string is being duplicated. Unfortunately, the MZSettings.StagingServerAddress property is not currently resettable, and supporting RESET in the Q_PROPERTY would be a larger change. Let me know if think this hack is not acceptable.

Reply:
I think this is acceptable, this is primarily used during development only so we we shouldn't overcomplicate it.

Reply:
TIL about `RESET`! Agree, I don't think it's worth making changes to the settings macro for this one dev menu feature. Since our app constants aren't exposed to QML, do you think there are any good spots to add a CONST Q_PROPERTY somewhere to expose the default staging url?

Reply:
Changed the solution to not require the default staging URL in QML.